test case: 'let' with a comment

bbigras opened the pull request:
The problem seems to go away if the comment is removed.

output

```nix
Unknown let variation let
  nixpkgs = import <nixpkgs>;
  # some comment
in
with nixpkgs;
  stdenv.mkDerivation {
    name = "something";
    buildInputs = [];
  }
```

bbigras committed:
test case: 'let' with a comment

justinwoo commented:
i don't know what i was expecting here, i really honestly thought that the expression items before `in` would go into a list of `binds`, but this was not the case? https://github.com/justinwoo/format-nix/blob/49867e6f9714370fee2ddfb32993d5ae29b94729/src/FormatNix.purs#L160-L163

grammar rule: https://github.com/cstrahan/tree-sitter-nix/blob/6c442630c428b52b1e16ae477effd0ae4ad46245/grammar.js#L92

justinwoo commented:
From the test output:

```
(expression
 (let
   (binds (bind (attrpath (identifier)) (app (identifier) (spath))))
   (comment)
   (with ...)))
```

justinwoo commented:
it might unfortunately be that you have to use the method `children` which gets annoymous children, which should then give you the nodes:

`let` `binds` `comment` `in` `with`

so if you drop the first and split the rest of the list by `in`, you should be able to get what is basically a `binds + extras` list and `rest` split

justinwoo committed:
handle let bindings with comments

one apparently cannot guarantee that within a let block, we will
actually correctly only have bindings. we may need to flatten "Binds"
in the future.

bbigras commented:
Thanks!!
